app/assets/javascripts/admin/order_cycles/services/order_cycle.js.coffee

```diff
@@ -156,7 +156,7 @@ angular.module('admin.orderCycles').factory 'OrderCycle', ($resource, $window, S
         if response.data.errors?
           StatusMessage.display('failure', response.data.errors[0])
         else
-          StatusMessage.display('failure', 'Failed to create order cycle')
+          StatusMessage.display('failure', t('js.order_cycles.create_failure'))
```

Comment:
I am not a big fan of this js namespace in en.yml.
I don't think we have a convention and we have a mix of js. entries and non-js entries being used in angular templates.
I think we need to define a convention and document it here:
https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-%28i18n%29

I'd vote for not using the js namespace and do what's done in admin.subscriptions for example.
Basically, I think these entries should be under admin.order_cycles together with the translations of that page, no matter if it's rails views or JS.

what do you guys think @kristinalim @mkllnk ?

Reply:
I was trying to check if this separation between rails and js translations is encouraged or recommended in i18n-js, but I dont think it is.

Reply:
The idea of the js namespace was that it's clear that these keys are used in Javascript. If we had a complete js namespace, we wouldn't need to include the whole locale and all translations in the assets. We could reduce the size of the assets a lot.

If we decide that it's unrealistic to separate these namespaces, we can choose another namespace. If I understand your comment, you are suggesting `admin.orderCycles.create_failure` here, right?

I definitely agree that we should standardise it so that we all go in the same direction. Thank you for bringing it up.

Reply:
yes, I am suggesting "admin.order_cycles.create_failure" which is together with the rest of the translations for this page. Or even "admin.order_cycles.form.create_failure" or "admin.order_cycles.edit.create_failure".

I was not aware of that advantage of the js namespace. Size of assets. It's a very good point!
Do you know how to split the files so the assets pipeline only picks up the js namespace? If that is possible, I'd change my vote actually. It'd be great to remove most translations from the assets.
So, I am changing my vote to agree with this commit and make it a convention. We can create issues to fix the main problems, for example, subscriptions. And we can create a "project" (maybe a wiki page?), to separate and remove most translations from assets.

Do you guys agree?

Reply:
before we invest any time in it, do we know how much time reduction we get precompiling the assets?

Reply:
@luisramos0 

> Do you know how to split the files so the assets pipeline only picks up the js namespace?

Yes. It's configurable: https://github.com/fnando/i18n-js#export-configuration-for-translations

@sauloperez 
> do we know how much time reduction we get precompiling the assets?

No. And I don't think it would be much. We were more discussing the speed-up for clients which don't have to download all translations. I reckon that is several hundred kb.

Reply:
yes, the advantage is smaller assets for the client.
Locally, I have 4 languages enabled, I copy/pasted the I18n.translations line on all.js to a file:   768kb :-)

Reply:
Nice point about reducing file size, @mkllnk. @luisramos0 I agree that it should be a convention.

Not sure @luisramos0 if you were talking about gradually separating the translations. If yes, IMO it would be most efficient to do the separation in one go. It sounds like a lot of testing needed otherwise, or a risk that it will take a really long time until we can benefit from the client-side just using its own namespace.

Reply:
Hmm... I'm wondering again about the download size, if there would be that big of a difference if the production server has Gzip compression enabled... But even then, browser memory would be very happy. :grinning:

Reply:
yes, I am ok with any implementation approach actually :-)

I checked the sizes a bit better. Just the en translations are 190k on the client, 2600 translations. It's ~7kb per 100 translations.
We have ~200 lines on js: right now, let's say this will move to 400 or 600 lines after migration. That will be 30kb or 40kb total.
We are talking about reducing 150kb (per active language in the instance).
We have to do this!

Reply:
good point. I zipped the 190kb file, I get 55kb :-)
according to this, we would still get ~40kb per language.

Reply:
Let's do it!